cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -387,6 +394,7 @@ class reader::impl {
 
   file_intermediate_data _file_itm_data;
   bool _file_preprocessed{false};
+  bool _first_pass_in_empty_table{false};
```

Comment:
I think this can just be "is first pass". We do always return at least one chunk.
That said, I wonder if we can derive if we're in the first pass from the existing state.

Reply:
I think we should keep it as `_first_pass_in_empty_table` as this variable only activates for empty tables to activate special handling (true until `read_chunk() called) of first pass in `has_next()` for empty tables. For non-empty tables, `has_next()` isn't affected by the pass number so this variable stays dormant.